lib/output.js

```diff
@@ -694,6 +694,7 @@ function OutputStream(options) {
         output.semicolon();
     });
     DEFPRINT(AST_Toplevel, function(self, output){
+        use_asm = false;
```

Comment:
Originally "use asm" logic used figure_out_scope() before outputting the code because it needed the raw token to decide how to format numbers as per asm.js spec. mishoo did not like output.js relying on figure_out_scope() so you that's why you see all that use_asm logic in output. The theory was that use_asm did not have to be reset because it would always be balanced. I think I had the same question at the time.